keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,696 @@
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: Editing the existing component config file to add the new minimum required field (metadata.name) and restarting the component.
+    - Will enabling / disabling the feature require downtime of the control
+      plane? 
+    - Will enabling / disabling the feature require downtime or reprovisioning
+      of a node? (Do not assume `Dynamic Kubelet Config` feature is enabled).
+
+* **Does enabling the feature change any default behavior?**
+  Any change of default behavior may be surprising to users or break existing
+  automations, so be extremely careful here.
+
+ No, these new fields don't control component behavior. There may be some impact to existing tests making use of previously overloaded component configs with fields of the same name if existing values are out of bounds.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+  Describe the consequences on existing workloads (e.g. if this is runtime
+  feature, can it break the existing applications?).
+
+Yes, this can be disabled once it has been enabled. The existing component API's have a lenient path implementation for the strict decoders that allow for previously overloaded configs with fields of the same name to pass validation.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+Nothing right now as decoders for the component configs provide a lenient path to allow for the fileds to be included or excluded.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+- NA
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+- There should be no chance of failure while the lenient path for the strict decoder is in place as the filed will be optional.
```

Comment:
Also here, we can state it's just metadata for tooling that shouldn't affect the runtime behavior, beyond increasing compatibility sometime in the future when we enable strict decoders.